twenty-first/src/math/bfield_codec.rs

```diff
@@ -732,58 +762,60 @@ mod tests {
         prop_assert_eq!(encoded, poly_w_leading_zeros_encoded);
     }
 
-    #[proptest]
-    fn disallow_trailing_zeros_in_poly_encoding_bfe(
-        mut polynomial: Polynomial<BFieldElement>,
-        #[filter(!#leading_coefficient.is_zero())] leading_coefficient: BFieldElement,
-        #[strategy(1usize..30)] num_leading_zeros: usize,
+    fn disallow_trailing_zeros_in_poly_encoding_prop<T: FiniteField + BFieldCodec>(
+        mut polynomial: Polynomial<T>,
+        leading_coefficient: T,
+        num_leading_zeros: usize,
     ) {
```

Comment:
I like this. If you make the return type a `PropTestResult`, you can even keep using `prop_assert_eq!` and `prop_assert!`.
```suggestion
    fn disallow_trailing_zeros_in_poly_encoding_prop<T: FiniteField + BFieldCodec>(
        mut polynomial: Polynomial<T>,
        leading_coefficient: T,
        num_leading_zeros: usize,
    ) -> TestCaseResult {
```

It would also require
- returning `Ok(())` in this function, and
- changing the two `proptest`s that use this function to conclude with `?;`.

Reply:
Done